src/MassTransit/Logging/Diagnostics/LogContextActivityExtensions.cs

```diff
@@ -266,7 +266,24 @@ static ActivityContext GetParentActivityContext(Headers headers, bool isRemote =
             activity.SetTag(DiagnosticHeaders.Messaging.Operation, "process");
 
             if (activity.IsAllDataRequested)
+            {
+                if (context.MessageId.HasValue)
+                    activity.SetTag(DiagnosticHeaders.MessageId, context.MessageId.Value.ToString("D"));
```

Comment:
I believe `DiagnosticHeaders.MessageId` is `messaging.masstransit.message_id`, why not use `messaging.message.id` attribute instead?

Reply:
I agree with you, but as requested, I aligned with the send activity that set `messaging.masstransit.message_id`. @phatboyg can you please share your thoughts?

Reply:
`messaging.message-id` is for the transport message ID, not the MassTransit generated MessageId (Guid).

Reply:
`context.MessageId` always contain the MT generated ID, right? Is there any way to get the transport message ID, so we can set `messaging.message.id`?

Reply:
It already sets it.
https://github.com/MassTransit/MassTransit/blob/84353adedff1bd212bf98e0cb521fe1c6832b475/src/MassTransit/Logging/Diagnostics/LogContextActivityExtensions.cs#L84